code/game/machinery/kitchen/microwave.dm

```diff
@@ -154,72 +154,81 @@
-				items_measures_p[display_name] = "fillets of meat"
-			items_counts[display_name]++
-		for (var/O in items_counts)
-			var/N = items_counts[O]
-			if (!(O in items_measures))
-				dat += {"<B>[capitalize(O)]:</B> [N] [lowertext(O)]\s<BR>"}
-			else
-				if (N==1)
-					dat += {"<B>[capitalize(O)]:</B> [N] [items_measures[O]]<BR>"}
-				else
-					dat += {"<B>[capitalize(O)]:</B> [N] [items_measures_p[O]]<BR>"}
-
-		for (var/datum/reagent/R in reagents.reagent_list)
-			var/display_name = R.name
-			if (R.id == "hotsauce")
-				display_name = "Hotsauce"
-			if (R.id == "frostoil")
-				display_name = "Coldsauce"
-			dat += {"<B>[display_name]:</B> [R.volume] unit\s<BR>"}
-
-		if (items_counts.len==0 && reagents.reagent_list.len==0)
-			dat = {"<B>The microwave is empty</B><BR>"}
+	data["operating"] = src.operating
+	data["broken"] = src.broken
+	data["dirty"] = (src.dirty == 100)
+
+	var/list/ingredients = new
+	var/list/items_counts = new
+	var/list/items_measures = new
+	var/list/items_measures_p = new
+
+	for (var/obj/O in contents)
+		var/display_name = O.name
+
+		if (istype(O,/obj/item/reagent_container/food/snacks/egg))
+			items_measures[display_name] = "egg"
+			items_measures_p[display_name] = "eggs"
+		if (istype(O,/obj/item/reagent_container/food/snacks/tofu))
+			items_measures[display_name] = "tofu chunk"
+			items_measures_p[display_name] = "tofu chunks"
+		if (istype(O,/obj/item/reagent_container/food/snacks/meat)) //any meat
+			items_measures[display_name] = "slab of meat"
+			items_measures_p[display_name] = "slabs of meat"
+		if (istype(O,/obj/item/reagent_container/food/snacks/donkpocket))
+			display_name = "Turnovers"
+			items_measures[display_name] = "turnover"
+			items_measures_p[display_name] = "turnovers"
+		if (istype(O,/obj/item/reagent_container/food/snacks/carpmeat))
+			items_measures[display_name] = "fillet of meat"
+			items_measures_p[display_name] = "fillets of meat"
+		items_counts[display_name]++
+
+	for (var/O in items_counts)
+		var/list/item = new
+
+		item["name"] = capitalize(O)
+		item["count"] = items_counts[O]
+
+		if (!(O in items_measures))
+			item["measure"] = "[lowertext(O)][items_counts[O] > 1 ? "s" : ""]" // Adds 's' for plurals.
+		else if (items_counts[O] == 1)
+			item["measure"] = items_measures[O]
 		else
-			dat = {"<b>Ingredients:</b><br>[dat]"}
-		dat += {"<HR><BR>\
-<A href='?src=\ref[src];action=cook'>Turn on!<BR>\
-<A href='?src=\ref[src];action=dispose'>Eject ingredients!<BR>\
-"}
+			item["measure"] = items_measures_p[O]
 
-	show_browser(user, dat, "Microwave Controls", "microwave")
-	return
+		ingredients += list(item)
 
+	for (var/datum/reagent/R in reagents.reagent_list)
+		var/list/reagent = new
 
+		reagent["count"] = R.volume
+		reagent["measure"] = R.volume > 1 ? "units" : "unit"
+
+		reagent["name"] = R.name
+		if (R.id == "hotsauce")
+			reagent["name"] = "Hotsauce"
+		if (R.id == "frostoil")
+			reagent["name"] = "Coldsauce"
```

Comment:
this seems kind of gross.

Reply:
Most of it is just copy pasted from the old code. The alternative is to have it called 'Frost oil' instead, I can remove it if you want.

Reply:
it can be left